include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
> `RenderLayers` should never outlive their associated `Layer`

Just realized this isn't the case. `RenderLayer`s copied into the worker may outlive the `Layer`. Need to think about this.